src/extension.ts

```diff
@@ -154,7 +154,8 @@ export async function activate(context: ExtensionContext) {
         uri,
         outputChannel,
         literal,
-        panel
+        panel,
+        combinedEnv
```

Comment:
This is the fix, which consists of passing the env variables imported from .env file. Not doing so will result in defaulting to process.env, which afaiu is different from the process of the server.
https://github.com/prisma/graphql-config/blob/master/src/extensions/endpoints/EndpointsExtension.ts#L65-L69